source/patterns/@aws-solutions-constructs/core/lib/cognito-helper.ts

```diff
@@ -120,3 +120,18 @@ export function setupCognitoForElasticSearch(scope: Construct, domainName: strin
 
   return cognitoAuthorizedRole;
 }
+
+export function setupOpenSearchCognito(scope: Construct, domainName: string):
+  [cognito.UserPool, cognito.UserPoolClient, cognito.CfnIdentityPool, iam.Role] {
+  const userPool = buildUserPool(scope);
+  const userPoolClient = buildUserPoolClient(scope, userPool);
+  const identityPool = buildIdentityPool(scope, userPool, userPoolClient);
+
+  const cognitoAuthorizedRole: iam.Role = setupCognitoForElasticSearch(scope, domainName, {
```

Comment:
Really? Calling setupCognitoForElasticSearch? We can't do that. I see the action in the role created is 'es:ESHttp*', did AWS really keep the ES prefixes for actions when writing OpenSearch?

If the two use cases can truly use the same code, let's get change the name to setupCognitoForSearchService

Reply:
AWS kept ES prefixes for OpenSearch